v3.0, v3.1, dev: add missed PRs in 3.0.6 (#1679)
pingcap/docs: dev/releases/3.0.6.md

    - Fix the issue that the query result is incorrect when the query statement contains a variable assignment expression [#13231](https://github.com/pingcap/tidb/pull/13231)
    - Fix the issue that the result is incorrect when the `UPDATE` statement contains both a sub-query and a generated column; fix the `UPDATE` statement execution error when this statement contains two same-named tables from different source databases [#13350](https://github.com/pingcap/tidb/pull/13350)
    - Support `_tidb_rowid` for point queries [#13416](https://github.com/pingcap/tidb/pull/13416)

>     - Fix the issue that query execution plan is incorrect when the partition table statistics is wrongly used [#13628](https://github.com/pingcap/tidb/pull/13628)

Reviewer: ```suggestion
    - Fix the issue that the generated query execution plan is incorrect, caused by incorrect usage of partitioned table statistics [#13628](https://github.com/pingcap/tidb/pull/13628)
```

    - Use the table’s `COLLATE` instead of the system’s default charset in the column when a table is created and the table contains `COLLATE` [#13174](https://github.com/pingcap/tidb/pull/13174)
    - Limit the length of the index name when creating a table [#13310](https://github.com/pingcap/tidb/pull/13310)
    - Fix the issue that the table name length is not checked when a table is renamed [#13346](https://github.com/pingcap/tidb/pull/13346)

>     - Add the new config `alter-primary-key` to support add/drop primary key on TiDB, this config is disable by default [#13522](https://github.com/pingcap/tidb/pull/13522)

Reviewer: ```suggestion
    - Add the `alter-primary-key` configuration (disabled by default) to support adding/dropping the primary key in TiDB [#13522](https://github.com/pingcap/tidb/pull/13522)
```

- Add the configuration check for Titan [#5720](https://github.com/tikv/tikv/pull/5720)
- Support using tikv-ctl to dynamically modify the GC I/O limit: `tikv-ctl --host=ip:port modify-tikv-config -m server -n gc.max_write_bytes_per_sec -v 10MB` [#5957](https://github.com/tikv/tikv/pull/5957)
- Reduce useless `clean up` requests to decrease the pressure on the deadlock detector [#5965](https://github.com/tikv/tikv/pull/5965)

> - Avoid reducing TTL in pessimistic lock prewrite request [#6056](https://github.com/tikv/tikv/pull/6056)

Reviewer: ```suggestion
- Avoid reducing TTL in pessimistic locking prewrite requests [#6056](https://github.com/tikv/tikv/pull/6056)
```

> - Fix the issue that may occur missing blob file in Titan [#5968](https://github.com/tikv/tikv/pull/5968)

Reviewer: ```suggestion
- Fix the issue that a missing blob file might occur in Titan [#5968](https://github.com/tikv/tikv/pull/5968)
```

> - Fix the issue that `RocksDBOptions` may not take effect in Titan [#6009](https://github.com/tikv/tikv/pull/6009)

Reviewer: ```suggestion
- Fix the issue that `RocksDBOptions` might not take effect in Titan [#6009](https://github.com/tikv/tikv/pull/6009)
```

## PD

- Add an `ActOn` dimension for each filter to indicate that each scheduler and checker is affected by the filter, and delete two unused filters: `disconnectFilter` and `rejectLeaderFilter` [#1911](https://github.com/pingcap/pd/pull/1911)
- Print a warning log when it takes more than 5 milliseconds to generate a timestamp in PD [#1867](https://github.com/pingcap/pd/pull/1867)
- Lower the client log level when passing unavailable endpoint to the client [#1856](https://github.com/pingcap/pd/pull/1856)

> - Fix the issue that RPC receive message may exceed max size in `RegionSyncer` replication [#1952](https://github.com/pingcap/pd/pull/1952)

Reviewer: ```suggestion
- Fix the issue that the gRPC message package might exceed the maximum size in the `region_syncer` replication process [#1952](https://github.com/pingcap/pd/pull/1952)
```
